platforms/core-configuration/configuration-cache/src/integTest/groovy/org/gradle/configurationcache/inputs/undeclared/AbstractUndeclaredBuildInputsIntegrationTest.groovy

```diff
@@ -305,6 +308,13 @@ abstract class AbstractUndeclaredBuildInputsIntegrationTest extends AbstractConf
         "constructing a byte channel with open options" | (TestFile it) -> { UndeclaredFileAccess.filesNewByteChannelWithOpenOptions(testFilePath(it)) }
         "reading lines from a file"                     | (TestFile it) -> { UndeclaredFileAccess.fileReadLines(testFilePath(it)) }
     }
+
+    void waitForChangesToBePickedUp() {
+        FileSystemWatchingHelper.waitForChangesToBePickedUp()
+        // the test is flaky, trying harder
+        FileSystemWatchingHelper.waitForChangesToBePickedUp()
+    }
```

Comment:
:question: Does this mean that for configuration our regular wait for changes to be picked up is not enough?

I don't think doubling the time like this is a good solution, though.

Reply:
Is there a reason why we must fix this test in this PR? Let's move that work out to a separate PR, and maybe base this one on that?

Reply:
Well, it's a small fix that manifested in my PR, I don't see value in having a separate PR for that, a commit is enough.

As for flakiness - I didn't investigate that deep. The test was failing with a single delay, so I doubled it.